### ISSUE TITLE
puppet-lint fails syntax check for valid variable types

### ISSUE THREAD

**Issue by nrvale0** [rodjek/puppet-lint#518]
puppet-lint incorrectly fails syntax check of files with Puppet4 Types as class parameters.

For instance:

``` puppet
class foo(
  Pattern[/^.+$/] $bar
) {
...
```

``` shellsession
ERROR: Syntax error (try running `puppet parser validate <file>`) on line 2
```

``` shellsession
$ puppet parser validate <path to file>
$ echo $?
0
```

**Comment by nrvale0** [comment]
This was because puppet-lint was picking up system Puppet (early 4.x) instead of the puppet I had installed in test environment using 'bundle exec'. Removing system Puppet resolved the issue in my environment but it seems that there is a lingering problem with honoring the user's path when puppet-lint tries to do a syntax check.

**Comment by alexjurkiewicz** [comment]
Actually, puppet-lint fails on the caret in `Pattern[/^/]`, not just any old variable type (eg try your file with `String[1]`). Could you please change this issue title to something like "puppet-lint doesn't parse variable types correctly"?

**Comment by nrvale0** [comment]
@alexjurkiewicz Subject updated.

BTW, where you able to figure out it was choking on the regex using standard command-line params or did you have to drop into debugger? I don't recall seeing an clues which would have led me to the regex specifically.

**Comment by alexjurkiewicz** [comment]
I'm starting to use variable types heavily, and I noticed only certain pattern types caused failures

**Comment by paran1** [comment]
puppet-lint 2.0.1 (and later) does not give syntax error for the example manifest. This was probably fixed by d7cdce6680eaee13a2d8eaf9d98c347577c07dc8.

**Comment by rnelson0** [comment]
@nrvale0 Closing this ticket as I confirm the sample has since been fixed. If you find other examples that fail, please open a new issue with more details. Thanks!

```
[rnelson0@build03 puppet-lint:master±]$ cat > 518.pp
class foo(
  Pattern[/^.+$/] $bar
) { }
[rnelson0@build03 puppet-lint:master±]$ puppet-lint --version
puppet-lint 2.0.2
[rnelson0@build03 puppet-lint:master±]$ puppet-lint 518.pp
ERROR: foo not in autoload module layout on line 1
WARNING: class not documented on line 1
```

**Comment by rnelson0** [comment]
Duplicate of #452
